clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java

```diff
@@ -88,33 +94,43 @@ void run(Collection<Provider> providers) {
        .filter(s -> !s.equals("_ALL_"))
        .collect(Collectors.toList());
 
-     List existingOnDemandKeys;
+     Map<String, Response<Boolean>> existingOnDemandKeys = new HashMap<>();
      if (redisClientDelegate.supportsMultiKeyPipelines()) {
-       existingOnDemandKeys = redisClientDelegate.withMultiKeyPipeline(pipeline -> {
-         onDemandKeys.forEach(k -> pipeline.get(provider.getProviderName() + ":onDemand:attributes:" + k));
-         return pipeline.syncAndReturnAll();
+       redisClientDelegate.withMultiKeyPipeline(pipeline -> {
+         for (List<String> partition : Iterables.partition(onDemandKeys, redisCacheOptions.getMaxDelSize())) {
```

Comment:
Could hardcode this to something small for the time being if we'd like -- it's not particularly critical

Reply:
We currently have `maxDelSize` set to 2 which makes me think we can do away with pipelineing here entirely for simplicity's sake, given that this cleanup agent isn't time sensitive. Totally fine to leave as-is though, should be a nice improvement.

Reply:
I didn't look close enough to see what the maxes were set to in the various profiles ... it looked like `maxPipelineSize` was pretty high which is why I err'd towards `maxDelSize`.

I'll stick with it as-is and see how it goes.